src/main/java/edu/texas/social_computing/hospitals/HRPP.java

```diff
@@ -31,19 +31,28 @@ public static void run(HospitalTable hospitalTable, ResidentTable residentTable)
             // try to match that person
             matching = HRP.run(hospitalTable, residentTable, unmatchedQueue);
             // if matched -> good
-            if(matching.hasAssignment(ndResident)) {
+            if (matching.hasAssignment(ndResident)) {
                 continue;
             }
-        }
-    }
-
-        // Task 2
-        // if not matched
 
+            // Task 2
+            // if partners still not proximally matched
             // unmatch BOTH in the couple and add both back to the queue (now no one is dominant)
+            matching.unassign(ndResident);
+            matching.unassign(partner);
+
+            // ndResident should still be in queue, right???
+            unmatchedQueue.add(partner);
 
             // reset non-dominant partner's preference list
+            ndResident.resetPreferences();
 
             // start from dominant partner's next hospital preference (changing the rank pointer)
+            residentTable.incrementResidentRankProgress(partner);
+            partner.setPrefsByProgress(residentTable.getResidentRankProgress(partner));
 
+            // run again
+            matching = HRP.run(hospitalTable, residentTable, unmatchedQueue);
```

Comment:
I think down here we will need to run the proximity violations check again and push any new people into violatingResidentsQ. Thoughts?

Reply:
Agreed. Done.